.github/workflows/conda_build_and_test.yml

```
@@ -96,8 +103,68 @@ jobs:
       - name: Build Triton
         run: |
           cd python
-          conda run --no-capture-output -n triton pip install wheel
-          conda run --no-capture-output -n triton pip install --no-build-isolation '.[tests]'
+          conda activate triton
+          pip install --no-build-isolation '.[tests]'
+
+      - name: Run core tests
+        run: |
+          conda activate triton
+          pip install pytest pytest-xdist
+          cd python/test/unit
+          python3 -m pytest --junitxml=~/reports/language.xml -n 8 --verbose --device xpu language/ --ignore=language/test_line_info.py
+          # run runtime tests serially to avoid race condition with cache handling.
+          python3 -m pytest --junitxml=~/reports/runtime.xml --device xpu runtime/
+          # run test_line_info.py separately with TRITON_DISABLE_LINE_INFO=0
+          TRITON_DISABLE_LINE_INFO=0 python3 -m pytest --junitxml=~/reports/line_info.xml --verbose --device xpu language/test_line_info.py
+
+      - name: Clear cache
+        run: |
+          rm -rf ~/.triton
+
+      - name: Run interpreter tests
+        env:
+          # TRITON_INTERPRET: "1"
+          CUDA_VISIBLE_DEVICES: ""
+        run: |
+          cd python/test/unit
+          conda activate triton
+          python3 -m pytest --junitxml=~/reports/flash_attention.xml --verbose --device xpu operators/test_flash_attention.py
+
+      - name: Run partial operators tests
+        run: |
+          cd python/test/unit
+          conda activate triton
+          python3 -m pytest --junitxml=~/reports/operators.xml -n 8 --verbose --device xpu operators
+
+      - name: Regression tests
+        run: |
+          conda activate triton
+          pip install pytest-rerunfailures
+          cd python/test/regression
+          python3 -m pytest --junitxml=~/reports/regression.xml -vvv -s --device xpu . --reruns 10 --ignore=test_performance.py
+
+      - name: Run XPU python tests
+        run: |
+          cd python/test/backend/third_party_backends
+          conda activate triton
+          python3 -m pytest -n auto --verbose test_xpu_backend.py
+
+      - name: Run Tutorials
+        run: |
+          conda activate triton
+          pip install matplotlib pandas tabulate
+          cd python/tutorials
+          python3 01-vector-add.py
+          python3 02-fused-softmax.py
+          python3 03-matrix-multiplication.py
+          python3 04-low-memory-dropout.py
+          python3 05-layer-norm.py
+          python3 06-fused-attention.py
+          python3 07-math-functions.py
```

Please rebase your branch and make the same change as in `build_and_test`:
```suggestion
          python3 07-extern-functions.py
```

---

Rebased